transports/dns/src/lib.rs

```diff
@@ -749,40 +799,155 @@ mod tests {
                 .unwrap()
                 .await
             {
-                Err(Error::ResolveError(e)) => match e.kind() {
-                    ResolveErrorKind::Proto(ProtoError { kind, .. })
-                        if matches!(kind.as_ref(), ProtoErrorKind::NoRecordsFound { .. }) => {}
-                    _ => panic!("Unexpected DNS error: {e:?}"),
-                },
+                Err(Error::Dial(dial_errs)) => {
+                    if dial_errs
+                        .iter()
+                        .any(|sub| matches!(sub, Error::ResolveError(_)))
+                    {
+                        // We found a ResolveError in the aggregator’s dial array
+                    } else {
+                        panic!(
+                            "Expected at least one ResolveError(...) sub-error, got {dial_errs:?}"
+                        )
+                    }
+                }
                 Err(e) => panic!("Unexpected error: {e:?}"),
                 Ok(_) => panic!("Unexpected success."),
             }
         }
-
+    
+        // Common DNS config.
+        let config = ResolverConfig::quad9();
+        let opts = ResolverOpts::default();
+    
         #[cfg(feature = "async-std")]
         {
             // Be explicit about the resolver used. At least on github CI, TXT
             // type record lookups may not work with the system DNS resolver.
-            let config = ResolverConfig::quad9();
-            let opts = ResolverOpts::default();
-            async_std_crate::task::block_on(
-                async_std::Transport::custom(CustomTransport, config, opts).then(run),
+            let transport = async_std_crate::task::block_on(
+                async_std::Transport::custom(CustomTransport, config.clone(), opts.clone()),
             );
```

Comment:
See above comment. This logic can be moved into the `test_async_std` helper function.